### ISSUE TITLE
HTTPS support/documentation

### ISSUE THREAD

**Issue by Hong-Revo** [rexyai/RestRserve#143]
Hello,

Apologies if this is a noobish question. I notice that the readme states RestRserve supports HTTPS; however is there anywhere that the process of adding this is documented?

**Comment by dselivanov** [comment]
We have an example here - https://github.com/rexyai/RestRserve/blob/master/inst/examples/ssl/app.R.

Contribution in a form of article would be appreciated!

**Comment by blosloos** [comment]
Yes, a more detailed documentation to use TLS over https with RestRserve would be wonderful; plain http may not be preferred by everyone nowadays. I absolutely appreciate your package, and have been using it to set up a http POST - it works great, and is well documented! But I have been trying to make sense of examples/ssl for a while now - in vain, especially by struggling with (a) the many underlying Rserve configs (e.g., what's a QAP server, set by a different port?) and (b) CA certification via, e.g., certbot and Let's encrypt. Of course, documentation is always work ...

**Comment by dselivanov** [comment]
Yeah, encryption is a bit complicated topic. I tend to use load balancers (HAproxy or nginx) in front of RestRserve  which do SSL termination. Or in practice I much more frequently use AWS ELB which does SSL termination and then routes traffic within private network to right services.

So Rserve and RestRserve do support SSL/TLS, but in practice I never used this feature.

@blosloos if you want to add an article I can help to review. But I don't see myself adding comprehensive docs about https any time soon.

**Comment by blosloos** [comment]
Thanks for your reply and details! Yes, a short article would be great, and I would add one (i.e., how to set up TLS with RestRserve on a UNIX server on some port, using certbot and let's encrypt, maybe with some hints/links on how certificates and the private/public key concept works for signatures and encryption, lots of fragmented information otherwise) - similarly, if I had the time ...

**Comment by pkinif** [comment]
Hi @blosloos, @dselivanov,

I am trying to deploy my RestRserve API using HTTPS/docker-compose/nginx/certbot. Deploying on port 8080 is not a problem for me, but it gets stuck as soon as I want to integrate HTTPS. If I understand correctly, RestRserve does not work with port 453 but port 8483? Do you know if, since your previous exchange, new documentation has been published somewhere?

**Comment by dselivanov** [comment]
Could you please describe your architecture in more details? Which service is responsible for ssl offloading? Nginx? Typically I recommend delegate SSL offloading to loadbalancer such as nginx or haproxy and make restrserve work in plain http mode.

**Comment by pkinif** [comment]
Hi @dselivanov,

I have pushed the code of my current architecture on this Github repo: https://github.com/pkinif/deploy_rertRserve_api_https
I got my SSL certificate from Cloudflare (still valid), and I am using Nginx. I am documenting the API with swagger. So the goal is to have the swagger UI on the api.example.com/8080/doc URL.

I have tried deploying on port 8443 (main branch) and port 8080 (port_8080 branch). 

For the 8443 scenarios, I am doing the same as your example: https://github.com/rexyai/RestRserve/blob/master/inst/examples/ssl/app.R.
But neither HTTP nor HTTPS is working.

For the 8080 scenarios, I can access my API at http://api.example.com:8080/doc# but not on the HTTPS version.

Either I have an error with Nginx, or I do something wrong with the RestRserve configuration.
